src/huggingface_hub/hf_api.py

```diff
@@ -1909,8 +1910,8 @@ def create_commit(
             num_threads=num_threads,
         )
         commit_payload = prepare_commit_payload(
-            additions=additions_with_upload_mode,
-            deletions=deletions,
```

Comment:
Is the local `deletions` variable still useful?

Reply:
Yup true we don't really need it anymore. Only remaining case was to count this list but we don't need the `deletions` list to be actually defined to count them.
I made the change, thanks :+1: